Pinpoint.Core/PluginEngine.cs

```diff
@@ -13,34 +14,33 @@ public class PluginEngine
 
         public List<IPluginListener<IPlugin, object>> Listeners { get; } = new List<IPluginListener<IPlugin, object>>();
 
-        public void AddPlugin(IPlugin plugin)
+        public void AddPlugin(IPlugin toAdd)
         {
-            var prevSettings = AppSettings.GetOrDefault("plugins", new List<PluginMeta>());
-
             // Disallow duplicate plugins
-            if (Plugins.Contains(plugin))
+            if (Plugins.Contains(toAdd))
             {
                 return;
             }
-            
+
             // Don't add plugin if it fails to load
-            if (!plugin.TryLoad())
+            if (!toAdd.TryLoad())
             {
                 return;
             }
 
-            var pluginName = plugin.Meta.Name;
-            if (prevSettings.Any(m => m.Name.Equals(pluginName)))
+            var plugins = AppSettings.GetOrDefault("plugins", new EmptyPlugin[0]);
```

Comment:
What is the need for EmptyPlugin - seems like it would throw a NotImplementedException as soon as you type, right? Also, what does the square brackets after "new EmptyPlugin" mean, I've never seen that syntax before

Reply:
Can't deserialize to an interface type hence `EmptyPlugin`